staging/src/k8s.io/apiserver/pkg/audit/types.go

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package audit
+
+import (
+	auditinternal "k8s.io/apiserver/pkg/apis/audit"
+)
+
+type Backend interface {
+	// Returns a channel so send events into. Per audit ID it might be that LogAuditEvent is called up to two times.
+	Sink() chan<- *auditinternal.Event
```

Comment:
just reading top to bottom, but I thought we were making calls synchronously in and the sink implementation could choose whether to make them async